test/test_log.py

```diff
@@ -55,28 +55,25 @@ def test_default_logging(self):
         ops.log.setup_root_logging(self.backend)
 
         logger = logging.getLogger()
-        self.assertEqual(logger.level, logging.INFO)
+        self.assertEqual(logger.level, logging.DEBUG)
         self.assertIsInstance(logger.handlers[0], ops.log.JujuLogHandler)
 
         test_cases = [(
-            lambda: logger.critical('critical'), ('CRITICAL', 'critical')
+            lambda: logger.critical('critical'), [('CRITICAL', 'critical')]
         ), (
-            lambda: logger.error('error'), ('ERROR', 'error')
+            lambda: logger.error('error'), [('ERROR', 'error')]
         ), (
-            lambda: logger.warning('warning'), ('WARNING', 'warning')
+            lambda: logger.warning('warning'), [('WARNING', 'warning')]
         ), (
-            lambda: logger.info('info'), ('INFO', 'info')
+            lambda: logger.info('info'), [('INFO', 'info')]
+        ), (
+            lambda: logger.debug('debug'), [('DEBUG', 'debug')]
         )]
 
         for do, res in test_cases:
             do()
             calls = self.backend.calls(clear=True)
-            self.assertEqual(len(calls), 1)
-            self.assertEqual(calls[0], res)
-        else:
-            logger.debug('debug')
-            calls = self.backend.calls(clear=True)
-            self.assertEqual(len(calls), 0)
+            self.assertEqual(calls, res)
```

Comment:
Much better, thanks.

Reply:
yeah, len() is rarely what you want, because you can't tell what is wrong when it doesn't match.